src/FSharp.Data.GraphQL.Server.AspNetCore/Giraffe/HttpHandlers.fs

```diff
@@ -1,155 +1,268 @@
+                    | Some op ->
+                        if not (op.OperationType = Ast.Query) then
+                            logger.LogTrace "Document operation is not of type Query"
+                            return createParsedContent () |> OperationQuery
                         else
-                            ()
-                        return! result |> applyPlanExecutionResult
-                    | Some queryAsStr ->
-                        let graphQLQueryDecodingResult =
-                            queryAsStr
-                            |> GraphQLQueryDecoding.decodeGraphQLQuery
-                                serializerOptions
-                                executor
-                                graphqlRequest.OperationName
-                                graphqlRequest.Variables
-                        match graphQLQueryDecodingResult with
-                        | Result.Error struct (docId, probDetails) ->
-                            return! httpOk cancellationToken interceptor serializerOptions (GQLResponse.RequestError (docId, probDetails)) next ctx
-                        | Ok query ->
-                            if logger.IsEnabled (LogLevel.Debug) then
-                                logger.LogDebug ($"Received query: %A{query}")
-                            else
-                                ()
-                            let root = rootFactory (ctx)
-                            let! result =
-                                let variables =
-                                    ImmutableDictionary.CreateRange (
-                                        query.Variables
-                                        |> Map.map (fun _ value -> JsonSerializer.SerializeToElement (value))
-                                    )
-                                executor.AsyncExecute (query.ExecutionPlan, data = root, variables = variables)
-                                |> Async.StartAsTask
-                            if logger.IsEnabled (LogLevel.Debug) then
-                                logger.LogDebug ($"Result metadata: %A{result.Metadata}")
+                            let hasNonMetaFields =
+                                Ast.containsFieldsBeyond
+                                    Ast.metaTypeFields
+                                    (fun x ->
+                                        logger.LogTrace($"Operation Selection in Field with name: {{fieldName}}", x.Name))
+                                    (fun _ -> logger.LogTrace "Operation Selection is non-Field type")
+                                    op
+
+                            if hasNonMetaFields then
+                                return createParsedContent() |> OperationQuery
                             else
-                                ()
-                            return! result |> applyPlanExecutionResult
-                }
-                if ctx.Request.Headers.ContentLength.GetValueOrDefault (0) = 0 then
```

Comment:
Is it enough to check the the request is empty? Maybe headers are not empty but body is?

Reply:
You addressed this remark yourself apparently. So, I'm resolving this conversation.